federation/schema.go

```diff
@@ -83,6 +92,50 @@ func convertVersionedSchemas(schemas serviceSchemas) (*SchemaWithFederationInfo,
 	fieldInfos := make(map[*graphql.Field]*FieldInfo)
 	for _, service := range serviceNames {
 		for _, typ := range serviceSchemasByName[service].Schema.Types {
+
+			// For federated fields parse the arguments to figure out which
+			// fields are the federated keys. They annotate that information
+			// on the field object.
+			if typ.Name == "Federation" {
+				for _, field := range typ.Fields {
+					for _, arg := range field.Args {
+
+						// Extract the type name from the formatting <object>-<service>
+						rootType := getRootType(arg.Type)
+						names := strings.SplitN(field.Name, "-", 2)
+						if len(names) != 2 {
+							return nil, oops.Errorf("Field %s doesnt have an object name and service name", field.Name)
+						}
+						objName := names[0]
+						inputType, ok := types[rootType.Name].(*graphql.InputObject)
+						if !ok {
+							return nil, oops.Errorf("Object %s is not an input object, but it is an argument to the field %s", rootType.Name, field.Name)
+						}
+
+						// Get the object type from the merged schema and loop through all it's fields
+						obj, ok := types[objName].(*graphql.Object)
+						if ok {
+							for fName, f := range obj.Fields {
+								// If the field is one of the input fields to the federatedfieldfunc,
+								// add the service name to the list of federated keys
+								if _, ok := inputType.InputFields[fName]; !ok {
```

Comment:
Should we also check if all `inputType.InputFields` have the corresponding `obj.Field`?

Reply:
We're looking through the obj.Fields in the line above, so we've already confirmed it is an field on the object